swarm/network/kademlia.go

```diff
@@ -336,24 +336,27 @@ func (k *Kademlia) EachBin(base []byte, pof pot.Pof, o int, eachBinFunc func(con
 	var i int
 	var startPo int
 	var endPo int
-	kadDepth := int(k.depth)
+	depth := k.neighbourhoodDepth()
 
 	k.conns.EachBin(base, pof, o, func(po, size int, f func(func(val pot.Val, i int) bool) bool) bool {
-		if po < kadDepth {
+		if po < depth {
 			endPo = po
 			if i > 0 {
 				startPo = endPo + 1
 			}
```

Comment:
there seems to be something wrong with this condition. What is its purpose? `i` in any case is always the same value as `po`, doesn't seem that's its intended use? in any case, always when `i > 0`, `startPo > endPo`, which results in no subscription for that `po` @zelig